tests/ui/cmp_owned/with_suggestion.rs

```diff
@@ -22,21 +24,15 @@ fn main() {
     "abc".chars().filter(|c| c.to_owned() != 'X');
 
     "abc".chars().filter(|c| *c != 'X');
-
-    let x = &Baz;
-    let y = &Baz;
-
-    y.to_owned() == *x;
-
-    let x = &&Baz;
-    let y = &Baz;
-
-    y.to_owned() == **x;
 }
 
 struct Foo;
 
 impl PartialEq for Foo {
+    // Allow this here, because it emits the lint
+    // without a suggestion. This is tested in
+    // `tests/ui/cmp_owned_without_suggestion.rs`
```

Comment:
NIT:
```suggestion
    // `tests/ui/cmp_owned/without_suggestion.rs`
```

r=me after that.